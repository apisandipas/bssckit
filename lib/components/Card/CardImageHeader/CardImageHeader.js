"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledTools = require("styled-tools");

var _theme = _interopRequireDefault(require("../../../theme"));

var _base = require("../../Utilities/base");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  width: 100%;\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    border-top-left-radius: ", ";\n    border-top-right-radius: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var getBorderRadius = function getBorderRadius(props) {
  if (props.noRadius) {
    return (0, _styledComponents.css)(_templateObject(), (0, _styledTools.theme)('cardImageHeader.borderRadius.noRadius'));
  }

  return (0, _styledComponents.css)(_templateObject2(), (0, _styledTools.theme)('cardImageHeader.borderRadius.topLeftRight'), (0, _styledTools.theme)('cardImageHeader.borderRadius.topLeftRight'));
};

var CardImageHeader = (0, _styledComponents["default"])(_base.Img)(_templateObject3(), function (props) {
  return getBorderRadius(props);
});
CardImageHeader.defaultProps = {
  theme: _theme["default"]
};
var _default = CardImageHeader;
exports["default"] = _default;