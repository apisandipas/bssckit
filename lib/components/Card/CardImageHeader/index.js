"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CardImageHeader", {
  enumerable: true,
  get: function get() {
    return _CardImageHeader["default"];
  }
});

var _CardImageHeader = _interopRequireDefault(require("./CardImageHeader"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }