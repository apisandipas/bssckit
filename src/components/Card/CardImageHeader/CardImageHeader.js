import styled, { css } from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../../theme';
import { Img } from '../../Utilities/base';

const getBorderRadius = props => {
  if (props.noRadius) {
    return css`
      border-radius: ${theme('cardImageHeader.borderRadius.noRadius')};
    `;
  }

  return css`
    border-top-left-radius: ${theme('cardImageHeader.borderRadius.topLeftRight')};
    border-top-right-radius: ${theme('cardImageHeader.borderRadius.topLeftRight')};
  `;
};

const CardImageHeader = styled(Img)`
  width: 100%;
  ${props => getBorderRadius(props)}
`;

CardImageHeader.defaultProps = {
  theme: defaultTheme,
};

export default CardImageHeader;
