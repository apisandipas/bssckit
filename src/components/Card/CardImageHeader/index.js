export { default as CardImageHeader } from './CardImageHeader';
