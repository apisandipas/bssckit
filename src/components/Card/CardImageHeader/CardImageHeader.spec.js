import React from 'react';
import { axe } from 'jest-axe';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import CardImageHeader from './CardImageHeader';

describe('CardImageHeader Component', () => {
  it('matches CardImageHeader', async () => {
    const { container } = render(<CardImageHeader alt='' />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('width', '100%');
    expect(container.firstChild).toHaveStyleRule('border-top-left-radius', 'calc(0.25rem - 1px)');
    expect(container.firstChild).toHaveStyleRule('border-top-right-radius', 'calc(0.25rem - 1px)');
  });

  it('matches CardImageHeader noRadius', () => {
    const { container } = render(<CardImageHeader noRadius />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-radius', '0');
  });
});
