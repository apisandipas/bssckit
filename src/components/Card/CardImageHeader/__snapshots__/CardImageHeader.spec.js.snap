// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImageHeader Component matches CardImageHeader 1`] = `
.c0 {
  width: 100%;
  border-top-left-radius: calc(0.25rem - 1px);
  border-top-right-radius: calc(0.25rem - 1px);
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<img
  alt=""
  class="sc-fznZeY c0"
/>
`;

exports[`CardImageHeader Component matches CardImageHeader noRadius 1`] = `
.c0 {
  width: 100%;
  border-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<img
  class="sc-fznZeY c0"
/>
`;
